### PR TITLE
Changing README and CONTRIBUTING (related to #401)

### PR DESCRIPTION
**Pull request by diyadas**
I've outlined the changes to the top of the README that I'd like to see for newcomers, as outlined in #401 > 2.

I also noted that there was a lot of contributing information in the README, which I found redundant and confusing with the actual CONTRIBUTING file, so I moved that to CONTRIBUTING. I also moved information about making PRs to one location in the CONTRIBUTING file, instead of split across 2. 

In a separate commit, I also removed the "What not to contribute" header -- prior to becoming in a maintainer, I have always glanced over this assuming that "what to contribute" aligned with my contribution. Imo, keeping that information in one section ensures that people are more likely to read it. We can skip this if you do not feel the same.

This is very experimental and I welcome discussion/edits to this PR. It was easier to make the changes as a proposal than to describe them exactly.

(and I have no idea why all these merge commits are coming up, sorry)

### CODE REVIEW

**Comment by diyadas** [comment]
@fmichonneau Happy New Year! Would you mind taking a look at this PR and the issue (#401) that led to it? This PR addresses part of the issue, but @rgaiacs indicated that you are in charge of changes like this. The changes to the issues template have already been merged into the styles repository.

**Comment by diyadas** [comment]
@katrinleinweber You brought up in #401 that the instructions for PRs are out of date, and additionally, I feel that it is confusing to have a half-summary on contributions in the README and not having people realize that they should read CONTRIBUTING.md because there are so many links? 

What would you say to updating the CONTRIBUTING.md to a better (single! instead of multiple) guide to PRs, and cutting the "Contributing" section of the README to:
> This lesson is improved by contributions from the community. We welcome both issues and pull requests. Please see our [CONTRIBUTING](https://github.com/swcarpentry/r-novice-inflammation/blob/master/CONTRIBUTING.md) document for information on how to make suggestions.

And everything else that needs to be preserved from that section being integrated into the CONTRIBUTING document?

**Comment by katrinleinweber** [comment]
I'd be fine with something like this: https://github.com/TIBHannover/BacDiveR/blob/master/README.md#how-to-contribute-see-contributingmd-file

**Comment by diyadas** [comment]
Apparently, I did that in November and it slipped my mind. Would you mind re-reviewing the CONTRIBUTING.md document? I made some significant changes to the instructions for submitting PRs. 

One outstanding issue with this is that the dependencies section of the lesson-example repo has been removed, and I don't know what to replace it with.

**Comment by katrinleinweber** [comment]
With "dependencies section … removed", do you mean https://github.com/carpentries/lesson-example/commit/52d301e01af3a8866757a5c99dfd7b9e1e6ba64c, https://github.com/carpentries/lesson-example/commit/8067917164b8e5b717772c4a5505c656d4936095, another commit, or [a particular issue there](https://github.com/carpentries/lesson-example/issues?utf8=%E2%9C%93&q=)?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -27,53 +32,49 @@ This is a good way to introduce yourself
 and to meet some of our community members.
 
 1.  If you do not have a [GitHub][github] account,
-    you can [send us comments by email][contact].
-    However,
-    we will be able to respond more quickly if you use one of the other methods described below.
+you can [send us comments by email][contact].
+However, we will be able to respond more quickly if you use one of the other 
+methods described below.
```
Comment (katrinleinweber) [inline CONTRIBUTING.md]: Starting here, large part of the diff seem to be "only" about line lengths, wrapping & indentation. I'm having trouble picking out the content changes and would prefer to keep at least the indentation. The [style guide doesn't mention it](https://www.google.com/search?q=indent+site:http://carpentries.github.io/lesson-example), but I'll add a commit to see whether the review is easier then.

**Comment by diyadas** [comment]
Thanks for the speedy review!

By dependencies, I'm referring to the link to: https://github.com/carpentries/lesson-example#dependencies I have no idea where that section is now, but I think: https://github.com/carpentries/lesson-example/blob/gh-pages/setup.md
addresses how to set up local previews ((Optional) Jekyll Setup for Lesson Development)/
I'm unclear what the link should be, though - I thought there was a nicer link?

I agree that it would be nice to match upstream repository instructions, but we're also on a different set of instructions due to the conversion to Travis, e.g. now issuing PRs against the master branch instead of gh-pages. 

(I'll merge after resolving the "how to generate local preview" issue.)

**Comment by diyadas** [comment]
Ah, https://carpentries.github.io/lesson-example/setup.html#optional-jekyll-setup-for-lesson-development is probably the link we want! Thoughts?

**Comment by katrinleinweber** [comment]
Got it, thanks! How about this?

Strange indeed where `## Depdendencies` went...

[Screen Shot 2019-05-02 at 19 42 33](https://github.com/carpentries/lesson-example/commits/gh-pages/README.md)

https://github.com/carpentries/lesson-example/commit/dece80d0999d48ebba94868a39f7a3a3e673fd30#diff-04c6e90faac2675aa89e2176d2eec7d8R124 still has it, but the _next_ commit (https://github.com/carpentries/lesson-example/commit/f2f424ec7cfc49a43c224b51009d730162e3eaf5#diff-04c6e90faac2675aa89e2176d2eec7d8) creates `README.md` anew?!? I think I'm too lazy to `git bisect` properly now :-D

**Comment by diyadas** [comment]
Looks great to me! Thanks for catching that bit of duplication. I really like how this has turned out :)

**Comment by katrinleinweber** [comment]
You're welcome! I like this outcome as well :-)

**Pull request merged**